### PR TITLE
Jameson game controller class

### PR DESCRIPTION
**Pull request by jameson28**
I've been putting a lot of work into the GameController Class and have made many updates that I would like you guys to have access to so I am doing this Pull Request so you can review what I have done.  Please take a few minutes to read over what I have done and let me know if any changes need to be made.  Don't just reply with "do it" or any other short message without substance.  Thanks!

### CODE REVIEW

**Comment by bhamor** [comment]
This looks pretty solid to me.  I did notice that your playMonopolyGame() method in the controller checks that there is still more than one player in the game, but doesn't have anything to deal with when there is only one left (endgame).  Is this something that will be added elsewhere or will it be added to this method later?  or did I just miss something?

File: MonopolyGame/src/cs414/a4/monopoly/backEnd/GameController.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+		return game.getPlayerNamesInOrder();
+	}
+	
+	public static void main(String [] args){
+		//Create view and get names from view
+		ViewController view = new ViewController();
+		List<String> playerNames = view.getPlayerNames();
+		
+		GameController gameController = new GameController(playerNames);
+		playerNames = gameController.getPlayerNames();
```
Comment (ajcopley) [inline MonopolyGame/src/cs414/a4/monopoly/backEnd/GameController.java]: Not sure why you are resetting playerNames here. You create the gameController with playerNames, but then overwrite playerNames with what you just set controller to. Maybe you are trying to get them to point to the same place in memory? Just a comment that i'm not sure what's going on here.

File: MonopolyGame/src/cs414/a4/monopoly/backEnd/Player.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
 	private boolean isInJail;
 	private int money;
 	private Bank bank;
-	private int turnsInJail;
+	protected int turnsInJail;
```
Comment (ajcopley) [inline MonopolyGame/src/cs414/a4/monopoly/backEnd/Player.java:15]: Maybe keep it private and use a getter and setter.

File: MonopolyGame/src/cs414/a4/monopoly/backEnd/Board.java [hunk 3]
```diff
@@ -44,6 +44,34 @@ public Board(List<Player> players) {
...
 		return players;
 	}
+	
+	public Property movePlayer(Player player, int distance){
+		player.movePlayer(distance);
+		return boardSpaces[player.getLocation()].getDeed();
+	}
+	
+	public void sellPropertyFromBankToPlayer(Property deed, Player player){
+		player.propertiesOwned.add(deed);
```
Comment (ajcopley) [inline MonopolyGame/src/cs414/a4/monopoly/backEnd/Board.java:54]: I believe all the lines player.propertiesOwned.add(deed) should be calling player.addProperty(property, cost) function. This checks to make sure the player can afford it. I believe the lines i say need replaced are directly manipulating the list and would need to have a accessor function. maybe discuss this with the team.

**Pull request closed without merging**